### PR TITLE
implement `wgpuComputePassEncoderSetPushConstants`

### PR DESCRIPTION
**Pull request by zackgomez**
It was straightforward to implement.  I don't expect my changes to the example to be merged, I am looking for guidance on whether they should be reverted, moved to another example, or something else entirely.

Test Plan:
Modified the compute shader to use push constants.  Verified the output.

### CODE REVIEW

**Comment by rajveermalviya** [comment]
Let's keep the `compute` example as-is, but a separate example for `push_constants` would be great to have. Also seems like `cargo fmt` CI is not happy.

**Comment by almarklein** [comment]
cc @fyellin since you implemented render push constants in wgpu-py.

**Comment by zackgomez** [comment]
rebased and added examples/push_constants

Also noting that I used my fork of wgpu with these PRs in a project and they all worked as expected

File: examples/push_constants/main.c [hunk 1]
```diff
@@ -0,0 +1,245 @@
+#include "framework.h"
+#include "webgpu-headers/webgpu.h"
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define LOG_PREFIX "[compute]"
```
Comment (rajveermalviya) [inline examples/push_constants/main.c]:
```suggestion
#define LOG_PREFIX "[push_constants]"
```

File: examples/push_constants/main.c [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  wgpuCommandEncoderCopyBufferToBuffer(command_encoder, storage_buffer, 0,
+                                       staging_buffer, 0, numbers_size);
+  
+  WGPUCommandBuffer command_buffer = wgpuCommandEncoderFinish(
+      command_encoder, &(const WGPUCommandBufferDescriptor){
+                           .label = "command_buffer",
+                       });
+  assert(command_buffer);
+
+  wgpuQueueWriteBuffer(queue, storage_buffer, 0, &numbers, numbers_size);
```
Comment (rajveermalviya) [inline examples/push_constants/main.c:226]: This write is unnecessary, correct?

**Review by rajveermalviya — CHANGES_REQUESTED** [review]
Looks great so far! Added couple of comments.

We could also consider porting over [this wgpu test](https://github.com/gfx-rs/wgpu/blob/trunk/tests/tests/push_constants.rs) so that we have a reference to track any future wgpu-core API changes and easily update the examples if needed.

**Comment by rajveermalviya** [comment]
Apart from the example, the API changes LGTM (and tests correctly with current example code). If you need we could merge the API changes in this PR, and do the example in a separate PR.

File: examples/push_constants/main.c (hunk 1 above)
Comment (zackgomez) [reply examples/push_constants/main.c]: applied

**Comment by fyellin** [comment]
When will this code be available to use in wgpu-py?

**Comment by almarklein** [comment]
> When will this code be available to use in wgpu-py?

@fyellin when a new version of wgpu-native is released, and wgpu-py updates to that release.

**Comment by zackgomez** [comment]
pls merge

**Review by rajveermalviya — APPROVED** [review]
Thanks, Looks good! 

(I also added a commit to run the example in CI)

**Comment by almarklein** [comment]
Thanks @zackgomez !

**Pull request merged**